compensations/src/test/java/org/jboss/narayana/compensations/functional/transaction/TransactionLocalTest.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2016, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.jboss.narayana.compensations.functional.transaction;
+
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.narayana.compensations.api.CompensationHandler;
+import org.jboss.narayana.compensations.api.ConfirmationHandler;
+import org.jboss.narayana.compensations.api.Transaction;
+import org.jboss.narayana.compensations.functional.common.DummyData;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import javax.inject.Inject;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+/**
+ * @author <a href="mailto:gytis@redhat.com">Gytis Trikleris</a>
+ */
+@RunWith(Arquillian.class)
+public class TransactionLocalTest {
+
+    @Inject
+    private Transaction transaction;
+
+    @Deployment
+    public static WebArchive createTestArchive() {
+        WebArchive archive = ShrinkWrap.create(WebArchive.class, "test.war")
+                .addClass(DummyData.class) // Deployment need at least one class with Compensations API annotations to get dependencies
+                .addAsWebInfResource(EmptyAsset.INSTANCE, "beans.xml");
+
+        return archive;
+    }
+
+    @Test
+    public void successfulScenario() {
+        final Counter actionsCounter = new Counter();
+        final Counter compensationsCounter = new Counter();
+        final Counter confirmationsCounter = new Counter();
+
+        transaction
```

Comment:
looks good - its why my questions about Action are ambiguous I don't see how these map to "Action" class